src/components/InputTimeInterval.vue

```diff
@@ -1,80 +1,91 @@
 <template lang="pug">
 div
   div
-    table
-      tr
-        th.pr-2
-          label(for="mode") Interval mode:
-        td
-          select(id="mode" v-model="mode")
-            option(value='last_duration') Last duration
-            option(value='range') Date range
-      tr(v-if="mode == 'last_duration'")
-        th.pr-2
-          label(for="duration") Show last:
-        td
-          select(id="duration" :value="duration", @change="valueChanged")
-            option(:value="15*60") 15min
-            option(:value="30*60") 30min
-            option(:value="60*60") 1h
-            option(:value="2*60*60") 2h
-            option(:value="4*60*60") 4h
-            option(:value="6*60*60") 6h
-            option(:value="12*60*60") 12h
-            option(:value="24*60*60") 24h
-      tr(v-if="mode == 'range'")
-        th.pr-2 Range:
-        td
-          input(type="date", v-model="start", @input="valueChanged")
-          input(type="date", v-model="end", @input="valueChanged")
+    div(class="settings-bar")
+      div
+        label(for="mode") Interval mode:
+        select(id="mode" v-model="mode")
+          option(value='last_duration') Last duration
+          option(value='range') Date range
+      div(v-if="mode == 'last_duration'")
+        label(for="duration") Show last:
+        select(id="duration" :value="duration" @change="changeDuration")
+          option(:value="15*60") 15min
+          option(:value="30*60") 30min
+          option(:value="60*60") 1h
+          option(:value="2*60*60") 2h
+          option(:value="4*60*60") 4h
+          option(:value="6*60*60") 6h
+          option(:value="12*60*60") 12h
+          option(:value="24*60*60") 24h
+      div(v-if="mode == 'range'")
+        | Range:
+        input(type="date", v-model="start")
+        input(type="date", v-model="end")
+      div
+        button.btn.btn-outline-dark(
+          type="button", 
+          :disabled="invalidDaterange || emptyDaterange",
+          @click="showTimeline"
+        ) Show Timeline
```

Comment:
I really don't think this button is necessary, especially with my suggested changes to `valueChanged`.

Reply:
I agree with not mentioning timeline in this component so that they are not coupled. I can make those changes. I included the button because in the issue, when looking at two date ranges (e.g. June 1 - June 2 and Apr. 1 - Apr. 2), after viewing the first date range, when you change June 1 to Apr. 1, aw-webui will render the timeline from Apr. 1 to Jun. 2, and the UI freezes for a couple of seconds while it is updating. Intuitively, I (and the issue author) expect the UI to wait for me to change both dates to Apr. 1 and Apr. 2 before updating the timeline. 

If I'm understanding you correctly, your suggested changes to valueChanged would still have the behaviour where the timeline updates before I finish inputting my second date, which is why I advocate for a button that sends an event up saying that the user has finished changing the date range. Let me know what you think!

Reply:
Yeah, I get you. Not really sure what to do about that, maybe show the button only when the `mode` is `range`?

Reply:
Yeah now I think that having the button only for range would look good. I think adding the `max-duration` prop would be a good idea too.

Reply:
I looked around to see if there's a configuration file to put `max-duration` in and it doesn't seem to be any except for Settings.vue, but I don't think users should be able to modify this value, should I just hard-code this value into InputTimeInterval.vue, Timeline.vue and Bucket.vue? 

By the way, I'm thinking of 31 days for `max-duration`. Thanks in advance for explaining all these little details to me!

Reply:
@ErikBjare I forgot to tag you, I didn't know that it wouldn't send you a notification

Reply:
Sorry for the delayed reply. I'd just make it a prop in `InputTimeInterval` with a default of null, and then set it to ~30 days from `Timeline.vue` and `Bucket.vue`).